cunumeric/deferred.py

```diff
@@ -95,24 +96,41 @@ def _prod(tpl: Sequence[int]) -> int:
 
 
 def auto_convert(
-    indices: Collection[int], keys: Sequence[str] = []
+    *thunk_params: str,
 ) -> Callable[[Callable[P, R]], Callable[P, R]]:
-    indices = set(indices)
+    """
+    Converts all named parameters to DeferredArrays.
+    """
+    keys: Set[str] = set(thunk_params)
 
     def decorator(func: Callable[P, R]) -> Callable[P, R]:
-        def wrapper(*args: Any, **kwargs: Any) -> Any:
+        assert not hasattr(
+            func, "__wrapped__"
+        ), "this decorator must be the innermost"
+
+        # For each parameter specified by name, also consider the case where
+        # it's passed as a positional parameter.
+        indices: Set[int] = set()
+        all_formals: Set[str] = set()
+        for (idx, param) in enumerate(signature(func).parameters):
+            all_formals.add(param)
+            if param in keys:
+                indices.add(idx)
+        assert len(keys - all_formals) == 0, "unkonwn parameter(s)"
```

Comment:
Since `all_formals` appends unconditionally every iteration, this assertions seems equivalent to `assert len(keys) == len(signature(func).parameters)`  in which case there is no need for `all_formals` at all.

Reply:
`keys` will likely be a subset of the wrapped function's arguments, and I'm also trying to catch the case where the user mis-spelled a parameter, e.g.:
```
@auto_convert("lhs")
def foo(self, rhs):
```
which is why I'm taking the set difference.

Reply:
Also I think
```
extra = keys - set(params)
assert len(extra) == 0, f"unknown parameter(s): {extra}"
```
will afford a more informative assertion message

Reply:
Good suggestions, thanks! Updated in 0b349c93f8d3a52699ecdc4ff3adc3d65246820d